File: app/src/main/java/me/imzack/app/ender/injector/module/TypeEditPresenterModule.java
package me.imzack.app.ender.injector.module;

import me.imzack.app.ender.view.contract.TypeEditViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class TypeEditPresenterModule {

    private final TypeEditViewContract mTypeEditViewContract;
    private final int mTypeListPosition;

    public TypeEditPresenterModule(TypeEditViewContract typeEditViewContract, int typeListPosition) {
        mTypeEditViewContract = typeEditViewContract;
        mTypeListPosition = typeListPosition;
    }

    @Provides
    TypeEditViewContract provideEditTypeViewContract() {
        return mTypeEditViewContract;
    }

    @Provides
    int provideTypeListPosition() {
        return mTypeListPosition;
    }
}

File: app/src/main/java/net/zackzhang/app/end/injector/module/TypeEditPresenterModule.kt
package net.zackzhang.app.end.injector.module

import dagger.Module
import dagger.Provides
import net.zackzhang.app.end.view.contract.TypeEditViewContract

@Module
class TypeEditPresenterModule(private val mTypeEditViewContract: TypeEditViewContract, private val mTypeListPosition: Int) {

    @Provides
    fun provideEditTypeViewContract() = mTypeEditViewContract

    @Provides
    fun provideTypeListPosition() = mTypeListPosition
}
